STAC-23376: Update Elasticsearch restore documentation (#97)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/data-management/backup_restore/kubernetes_backup.adoc

=== Delete Elasticsearch indices

[NOTE]
====
You can use the `--delete-all-indices` flag with the restore script to automatically delete all indices before restoring. See xref:/setup/data-management/backup_restore/kubernetes_backup.adoc#_restore_an_elasticsearch_snapshot[restore an Elasticsearch snapshot] for details.
====

> To manually delete existing Elasticsearch indices so that a snapshot can be restored, follow these steps.

Reviewer: ```suggestion
To manually delete the existing `Elasticsearch` indices and restore a snapshot, follow these steps:
```

> . Stop indexing - scale down all `*2es` deployments to 0:

Author: Thanks, I fixed that.

+
[,bash]
----

> kubectl scale --replicas=0 deployment -l observability.suse.com/scalable-during-es-restore="true"

Author: As far as I know, we don't keep the history of changes in the doc itself. I don't know what would be the best approach here.

====
*When a snapshot is restored, existing indices won't be overwritten.*

You can use the `--delete-all-indices` flag to automatically delete all indices by the restoring script, or manually delete them as described in xref:/setup/data-management/backup_restore/kubernetes_backup.adoc#_delete_elasticsearch_indices[delete Elasticsearch indices].

====

[WARNING]
====
*If the Elasticsearch PersistentVolumes were re-created* (for example, by accidental deletion and following pod restart), the Elasticsearch backup configuration must be recreated. This can be done by either:

* Reinstalling the SUSE Observability Helm chart with the same configuration used for the initial installation, or
* Manually triggering the backup initialization CronJob:
+
[,bash]
----
kubectl create job --from=cronjob.batch/suse-observability-backup-init "suse-observability-backup-init-$(date +%s)"
----
====
To restore an Elasticsearch snapshot, select a snapshot name and pass it as the first parameter. You can optionally specify:

> * A comma-separated list of indices to restore. If not specified, all indices matching the Helm value `backup.elasticsearch.scheduled.indices` will be restored. The default value is `"sts*"`).

Author: It is true,`.ds-sts_k8s_logs*` fall into that multi-target expression.

> * The `--delete-all-indices` flag to automatically delete all existing indices before restoring

Reviewer: ```suggestion
* The `--delete-all-indices` flag to automatically delete all existing indices before restoring.
```
